src/components/Shared/AddressTab.js

```diff
@@ -208,98 +209,69 @@ const AddressTab = ({
           </Grid>
 
           <Grid item xs={12}>
-            <CustomLookup
-              name='city'
-              label={labels.city}
-              required={requiredOptional ? false : true}
-              readOnly={(readOnly || !addressValidation.values.countryId) && true}
-              valueField='name'
-              displayField='name'
-              store={cityStore}
-              setStore={setCityStore}
-              onLookup={searchQry => {
-                setCityStore([])
-                if (!addressValidation.values.countryId) {
-                  console.log('false')
 
-                  return false
-                }
-                getRequest({
-                  extension: SystemRepository.City.snapshot,
-                  parameters: `${new URLSearchParams({
-                    _size: 30,
-                    _startAt: 0,
-                    _filter: searchQry,
-                    _countryId: addressValidation.values.countryId,
-                    _stateId: addressValidation.values.stateId ?? 0
-                  }).toString()}`
-                })
-                  .then(res => {
-                    console.log(res.list)
-                    setCityStore(res.list)
-                  })
-                  .catch(error => {
-                    setErrorMessage(error)
-                  })
-              }}
-              firstValue={addressValidation.values.city}
-              secondDisplayField={false}
-              onChange={(event, newValue) => {
-                if (newValue) {
-                  addressValidation.setFieldValue('cityId', newValue?.recordId)
-                  addressValidation.setFieldValue('city', newValue?.name)
-                } else {
-                  addressValidation.setFieldValue('cityId', null)
-                  addressValidation.setFieldValue('city', null)
-                }
-                addressValidation.setFieldValue('cityDistrictId', null)
-                addressValidation.setFieldValue('cityDistrict', null)
-              }}
-              error={addressValidation.touched.cityId && Boolean(addressValidation.errors.cityId)}
-              helperText={addressValidation.touched.cityId && addressValidation.errors.cityId}
-              maxAccess={maxAccess}
+            <ResourceLookup
+             endpointId={SystemRepository.City.snapshot}
+             parameters={{
+              _size: 30,
```

Comment:
SY.snapshotCIT requests only _stateId, _countryId and _filter
please remove the unnecessary parameters